libs/blocks/merch/merch.js

```diff
@@ -1,57 +1,151 @@
-import { loadScript, getConfig, createTag } from '../../utils/utils.js';
+import {
+  loadScript,
+  getConfig,
+  createTag,
+} from '../../utils/utils.js';
 
-const { miloLibs, codeRoot, env } = getConfig();
-const base = miloLibs || codeRoot;
+const wcs = { apiKey: 'wcms-commerce-ims-ro-user-cc' };
+const envProd = 'prod';
+const ctaPrefix = /^CTA +/;
 
-function buildButton(a, osi) {
-  if (!a) return null;
+let initialized = false;
+
+// supported languages for the price literals
+const supportedLanguages = [
+  'ar', 'bg', 'cs', 'da', 'de', 'en', 'es', 'et', 'fi', 'fr', 'he', 'hu', 'it', 'ja', 'ko',
+  'lt', 'lv', 'nb', 'nl', 'pl', 'pt', 'ro', 'ru', 'sk', 'sl', 'sv', 'tr', 'uk', 'zh_CN', 'zh_TW',
+];
+
+/**
+ * Removes undefined properties from an object
+ * @param {*} target Object with potentially undefined properties
+ * @returns new Object without undefined properties
+ */
+export function omitUndefined(target) {
+  if (target != null) {
+    Object.entries(target).forEach(([key, value]) => {
+      if (value == null) delete target[key];
+    });
+  }
+  return target;
+}
+
+const getTacocatEnv = (envName, ietf) => {
+  const scriptUrl = (envName === envProd
+    ? 'https://www.adobe.com/special/tacocat/lib/1.12.0/tacocat.js'
+    : 'https://www.stage.adobe.com/special/tacocat/lib/1.12.0/tacocat.js');
+  // eslint-disable-next-line prefer-const
+  let [language, country = 'us'] = ietf.split('-', 2);
+  if (!supportedLanguages.includes(language)) {
+    language = 'en'; // default to english
+  }
+  const literalScriptUrl = (envName === envProd
+    ? `https://www.adobe.com/special/tacocat/literals/${language}.js`
+    : `https://www.stage.adobe.com/special/tacocat/literals/${language}.js`);
+
+  return { scriptUrl, literalScriptUrl, country, language };
+};
+
+function initTacocat(envName, country, language) {
+  window.tacocat.tacocat({
+    defaults: {
+      country,
+      language,
+    },
+    environment: envName,
+    wcs,
+    literals: window.tacocat.literals[language],
+  });
+}
+
+function loadTacocat() {
+  if (initialized) {
+    return;
+  }
+  initialized = true;
+  const {
+    env,
+    locale: { ietf },
+  } = getConfig();
+  const { scriptUrl, literalScriptUrl, country, language } = getTacocatEnv(env.name, ietf);
+
+  Promise.all([
+    loadScript(literalScriptUrl, undefined, true),
```

Comment:
If the requested literals script is not available for some reason it will fail the whole tacocat price render right?
maybe we could have a fallback and try to load en literals instead?
or even have the hard version of en literals somewhere in the source code for the worst-case scenario.

if page has only CTA it doesn't even need literals, so taco for sure should load for it without literals js in place.

Reply:
en literals are bundled with consonant-templates in tacocat as a fallback. Current tacocat implementation requires the literals to be provided before initialisation even if empty object.

What you suggest could be an improvement on tacocat first and then we can update milo codebase.

I have improved the error handling so if the literal resource cannot be loaded, it will not block the initialization.